File: week01/Theme02/Theme02_Code03.py
# Theme01_Code02.py

import queue

q = queue.Queue()
    
q.put(0)
q.put(1)
q.put(2)

print(q.qsize(), q.empty()) # 3 False

q.put(3)
q.put(4)
q.get()
q.get()

print(q.qsize(), q.empty()) # 3 False

q.get()
q.get()
q.get()

print(q.qsize(), q.empty()) # 0 True

File: week01/Theme02/Theme02_Code03.cpp
// Theme02_Code03.cpp

#include <iostream>
#include <queue>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	queue<int> q;

	q.push(0);
	q.push(1);
	q.push(2);

	cout << q.size() << endl; // 3
	cout << q.empty() << endl; // 0
	
	q.push(3);
	q.push(4);
	q.pop();
	q.pop();

	cout << q.size() << endl; // 3
	cout << q.empty() << endl; // 0
	
	q.pop();
	q.pop();
	q.pop();
	
	cout << q.size() << endl; // 0
	cout << q.empty() << endl; // 1
}